exporters/common/src/main/java/io/opentelemetry/exporter/internal/grpc/GrpcExporterBuilder.java

```diff
@@ -207,8 +208,8 @@ public GrpcExporter<T> build() {
             grpcChannel,
             grpcStubFactory,
             retryPolicy,
-            tlsConfigHelper.getSslContext(),
-            tlsConfigHelper.getTrustManager());
+            isPlainHttp ? null : tlsConfigHelper.getSslContext(),
```

Comment:
> Since artifact layout is customized for Spring's custom Launcher classloader as mentioned above, JDK level security components are initialized without custom KeyStoreSpi implementation, because custom KeyStoreSpi implementation is not visible to system classloader, but to Spring's custom Launcher classloader.

It looks like the call to `SSLContext.getInstance("TLS")` is what is responsible for initializing this state? Seems brittle to need to rely on the otel java agent and otel java SDK to _not_ call ordinary java network APIs like `SSLContext.getInstance("TLS")`. 

Why is it not ok to call `SSLContext.getInstance("TLS")` when the scheme is `http`, but acceptable to call it when scheme is `https`?  Wouldn't spring users that want to implement `KeyStoreSpi` and use `https` be subject to this problem?

Reply:
Hi @jack-berg,

Actually, in my case, extra `SSLContext.getInstance("TLS")` call is not a problem. But I have though that there might be similar problem for `javax.net.ssl.SSLContextSpi` and `java.security.Provider` as they are also loaded by Java `ServiceLoader`. So I have prevented extra `SSLContext.getInstance("TLS")` calls if they are not needed (scheme is `http`). 

If you want, I can revert those changes and only keep 
```java
builder.connectionSpecs(Collections.singletonList(ConnectionSpec.CLEARTEXT));
```
parts as those fix the my original issue.

Reply:
I think this should be evaluated as a performance optimization where we try to avoid doing unnecessary work in obvious cases rather than creating a guarantee that it never happens.

While this has a similar appearance as java util logging which we take great pains to deal with in the javaagent, doing so adds a significant amount of complexity in the javaagent.  

For logging this complexity is worth it because logging initialization is so hard to avoid, while SSL initialization is only done in very specific locations and not difficult to avoid (with the added benefit of reducing performance overhead and not even requiring an additional setting).

Reply:
Actually, for the performance optimization perspective, getting rid of redundant `SSLContext.getInstance("TLS")` is not the only part optimized here. But also getting rid of redundant TrustManager and KeyStore initialization in the `OkHttpClient`

Reply:
@tylerbenson @jack-berg What you are thinking about this PR? Is there anything I can add/fix this PR? Or the Do changes make sense you?